### PR TITLE
sdk: Add mustache-c, bison and flex

### PR DESCRIPTION
**Pull request by smspillaz**
We're going to use mustache-c in T21272 in order to port
articleHTMLRenderer to C. Bison and Flex are required to
build mustache-c.

https://phabricator.endlessm.com/T21272

### CODE REVIEW

File: com.endlessm.apps.Sdk.json.in.in
Comment (ebassi) [inline com.endlessm.apps.Sdk.json.in.in]: This is kind of problematic: building from Git should only be reserved to components we control, since we also control the release schedule. Third party modules should either be built from a Git tag or from a tarball.

File: com.endlessm.apps.Sdk.json.in.in
Comment (ebassi) [inline com.endlessm.apps.Sdk.json.in.in]: This leaves `bison` and `flex` in the SDK, which is not something I'm really looking forward to.

In any case, though, both `bison` and `flex` are in the org.freedesktop.Sdk run time, so they should not be needed.

**Review by ebassi — CHANGES_REQUESTED** [review]

**Comment by ebassi** [comment]
Aside from the details of the change, I don't think adding a new dependency for legacy applications is a good plan. Can't we just vendor this into the content rendering library?

File: com.endlessm.apps.Sdk.json.in.in
Comment (smspillaz) [reply com.endlessm.apps.Sdk.json.in.in]: Fair enough. This dependency doesn't really have a release schedule to speak of, but I suppose we can just pin it to a certain commit (or create our own fork and pull from that).

**Comment by smspillaz** [comment]
> Can't we just vendor this into the content rendering library?

I'm not particularly interested in doing that - handling 3rd party dependencies like this is kind of what flatpak was designed for.

~~Of course, this is more of a transitional sort of thing. I was thinking that longer term we wouldn't have this in the SDK but the `eknrenderer` library and `mustache-c` library would be added as a separate `source` for both the SDK and for the Companion App Service (which will eventually stop being linked to the SDK itself due to the SDK compatibility drama).~~

I realised I contradicted myself mid-sentence there...

I think I'd rather have this in the SDK. Vendoring the dependency in knowledge-lib is just an uglier way of doing the same thing. If we don't want to expose it as part of the SDK, we could also just add `*` to `cleanup`.

In any event, we'll also have the same library as a `sources` entry to the Companion App Service when we build that flatpak, as it won't be linked to the SDK.

File: com.endlessm.apps.Sdk.json.in.in
Comment (smspillaz) [reply com.endlessm.apps.Sdk.json.in.in]: Fair enough. As long as mustache-c builds, I'm happy.

**Comment by ebassi** [comment]
> handling 3rd party dependencies like this is kind of what flatpak was designed for.

Sure, but then I get to maintain this, and get broken builds, or check for security issues. *Somebody* gets to maintain stuff every time we add a new dependency; it's not a free for all just because we can bundle stuff.

**Comment by smspillaz** [comment]
> Sure, but then I get to maintain this, and get broken builds, or check for security issues. Somebody gets to maintain stuff every time we add a new dependency; it's not a free for all just because we can bundle stuff.

Would that concern be ameliorated if we pinned the version? Its basically the same issue if you vendor the library.

**Comment by ebassi** [comment]
Yes, pinning to a specific version or commit would be the best option; I'd probably import this into a fork under endlessm, so we control it a bit better, and we can branch it alongside the rest of our stuff whenever we cut a new SDK release.

**Comment by smspillaz** [comment]
Okay, lets go with that then. Blocked until we have an `endlessm/` fork of mustache-c which we can separately version and use in the SDK manifest.

For what its worth, the upstream maintainer has been fairly responsive and co-operative with pull requests from me, so it'd be nice if we can make sure to upstream any changes we end up making.

File: com.endlessm.apps.Sdk.json.in.in
Comment (cosimoc) [reply com.endlessm.apps.Sdk.json.in.in]: Yeah, please pin it to a specific commit you know working, if you can't use a tag or tarball.

**Review by cosimoc — CHANGES_REQUESTED** [review]

**Review by cosimoc — CHANGES_REQUESTED** [review]
My comment still needs addressing

**Comment by smspillaz** [comment]
> My comment still needs addressing

Done. I wasn't sure whether we wanted to fork the project and manage the release schedule ourselves or use a specific git commit, but I've gone with the latter for now.

**Review by cosimoc — APPROVED** [review]
LGTM pending the other work.

**Review by ebassi — APPROVED** [review]

**Pull request merged**
